### PR TITLE
[pkg/pdatatest] Ensure all fields are checked

### LINKED ISSUE #17865: [internal/comparetest] Ensure all the fields are checked
Currently, some fields are ignored, e.g., many are missed in the number datapoints comparison. We need to ensure all data fields are validated and introduce explicit options to ignore some fields like timestamps.

### PR DESCRIPTION
**Pull request by dmitryax**
Resolves https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/17865

### CODE REVIEW

File: pkg/pdatatest/plogtest/logs_test.go [hunk 1]
```diff
@@ -172,7 +172,7 @@ func TestCompareLogs(t *testing.T) {
 			if tc.withOptions == nil {
 				assert.NoError(t, err)
 			} else {
-				assert.EqualError(t, tc.withOptions, err.Error())
+				assert.EqualError(t, err, tc.withOptions.Error())
```
Comment (dmitryax) [inline pkg/pdatatest/plogtest/logs_test.go:175]: Opportunistic fix. Expected value is the second argument in `EqualError`

File: pkg/pdatatest/plogtest/logs.go [hunk 2]
```diff
@@ -96,12 +96,7 @@ func CompareLogs(expected, actual plog.Logs, options ...CompareLogsOption) error
 // CompareResourceLogs compares each part of two given ResourceLogs and returns
 // an error if they don't match. The error describes what didn't match.
 func CompareResourceLogs(expected, actual plog.ResourceLogs) error {
-	var errs error
-
-	if !reflect.DeepEqual(expected.Resource().Attributes().AsRaw(), actual.Resource().Attributes().AsRaw()) {
-		errs = multierr.Append(errs, fmt.Errorf("attributes don't match expected: %v, actual: %v",
-			expected.Resource().Attributes().AsRaw(), actual.Resource().Attributes().AsRaw()))
-	}
+	errs := internal.CompareResource(expected.Resource(), actual.Resource())
```
Comment (fatsheep9146) [inline pkg/pdatatest/plogtest/logs.go:100]: Does `SchemaUrl` in `ResourceSpans`, `ResourceLogs`, `ResourceMetrics` should be compared in `CompareResourcexxx` functions? @dmitryax

Comment (dmitryax) [reply pkg/pdatatest/plogtest/logs.go:100]: Yeah, I also just found this out.

Comment (dmitryax) [reply pkg/pdatatest/plogtest/logs.go:103]: Updated. PTAL

**Review by djaglowski — APPROVED** [review]

File: pkg/pdatatest/internal/util.go [hunk 3]
```diff
@@ -35,3 +36,46 @@ func AddErrPrefix(prefix string, in error) error {
...
+	if expected.Version() != actual.Version() {
+		errs = multierr.Append(errs, fmt.Errorf("version doesn't match expected: %s, actual: %s",
+			expected.Version(), actual.Version()))
+	}
+	return multierr.Combine(errs,
+		CompareAttributes(expected.Attributes(), actual.Attributes()),
+		CompareDroppedAttributesCount(expected.DroppedAttributesCount(), actual.DroppedAttributesCount()))
+}
+
+func CompareSchemaUrl(expected, actual string) error {
```
Comment (djaglowski) [inline pkg/pdatatest/internal/util.go:62]:
```suggestion
func CompareSchemaURL(expected, actual string) error {
```

**Review by djaglowski — COMMENTED** [review]
Looks good to me, just needs a lint fix.

**Review by fatsheep9146 — APPROVED** [review]

**Pull request merged**